packages/yoroi-extension/app/containers/wallet/voting/VotingPage.js

```diff
@@ -102,24 +135,93 @@ export default class VotingPage extends Component<Props> {
         </VerticallyCenteredLayout>
       );
     }
-
     // keep enabled on the testnet
+    const catalystRoundInfo = this.generated.stores.substores.ada.votingStore.catalystRoundInfo;
     if (!environment.isTest()) {
-      const isLate = new Date() >= roundInfo.endDate;
-      const isEarly = new Date() <= roundInfo.startDate;
-      if (
-        selected.getParent().getNetworkInfo().NetworkId === networks.CardanoMainnet.NetworkId &&
-        (isEarly || isLate)
-      ) {
+      if (!catalystRoundInfo || (!catalystRoundInfo.currentFund && !catalystRoundInfo.nextFund)){
         return (
-          <RegistrationOver roundNumber={
-              isLate
-                ? roundInfo.nextRound
-                : roundInfo.nextRound - 1
-            }
+          <RegistrationOver
+            title={intl.formatMessage(messages.unavailableTitle)}
+            subtitle={intl.formatMessage(messages.unavailableSubtitle)}
           />
-        );
-      }
+          )
+        }
+        const { currentFund, nextFund } = catalystRoundInfo;
+        const nextFundRegistrationSubtitle = intl.formatMessage(messages.nextFundRegistration, {
+          roundNumber: nextFund?.id,
+          registrationStart: nextFund?.registrationStart
+        })
+
+        if(currentFund) {
+          const isLate = new Date() >= new Date(Date.parse(currentFund.registrationEnd))
+          const isEarly = new Date() <= new Date(Date.parse(currentFund.registrationStart))
+          const isBeforeVoting = new Date() <= new Date(Date.parse(currentFund.votingStart))
+          const isAfterVoting = new Date() >= new Date(Date.parse(currentFund.votingEnd))
+          const isBetweenVoting = !isBeforeVoting && !isAfterVoting;
+
+          if(isEarly) {
+            return (
+              <RegistrationOver
+                title={intl.formatMessage(messages.earlyForRegistrationTitle)}
+                subtitle={intl.formatMessage(messages.earlyForRegistrationSubTitle, {
+                  roundNumber: currentFund.id,
+                  registrationStart: currentFund.registrationStart
+                })}
+              />
+              )
+            }
+
+            // registeration is ended -> check for voting start and end dates
+            if (true || isLate) {
```

Comment:
Please check the condition

Reply:
Sorry 😄😄
I was testing that this case will give the expected response